src/components/Calendars/ClientScheduleAppointment.tsx

```diff
@@ -0,0 +1,432 @@
+        'Outubro',
+        'Novembro',
+        'Dezembro',
+    ];
+
+    const DAYS = ['Dom', 'Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sáb'];
+
+    const [month, setMonth] = useState(new Date().getMonth());
+    const [day, setDay] = useState(new Date().getDate());
+    const [year, setYear] = useState(new Date().getFullYear());
+    const [no_of_days, setNumDays] = useState<number[]>([]);
+    const [blankdays, setBlankDays] = useState<number[]>([]);
+    const [datepickerValue, setDatepickerValue] = useState('');
+    const [selectedHour, setSelectedHour] = useState<number | null>();
+
+    const [step, setStep] = useState(0);
+    const [availabilitiesMap, setAvailabilities] = useState<Map<string, number[]>>();
+    const [availableHours, setAvailableHours] = useState<number[]>();
+
+    // Calculates the number of days in the given month
+    const getNoOfDays = useCallback(
+        (month: number) => {
+            let daysInMonth = new Date(year, month + 1, 0).getDate();
+            let dayOfWeek = new Date(year, month).getDay();
+
+            let blankdaysArray: number[] = [];
+            for (var i = 1; i <= dayOfWeek; i++) {
+                blankdaysArray.push(i);
+            }
+
+            let daysArray: number[] = [];
+            for (var i = 1; i <= daysInMonth; i++) {
+                daysArray.push(i);
+            }
+
+            setBlankDays(blankdaysArray);
+            setNumDays(daysArray);
+        },
+        [year],
+    );
+
+    useEffect(() => {
+        getNoOfDays(month);
+    }, [month, getNoOfDays]);
+
+    useEffect(() => {
+        if (availabilities) {
+            const datesList = availabilities.map((d: Availability) => d.attributes.date);
+            const dateMap: Map<string, number[]> = new Map();
+
+            for (let d of datesList) {
+                const date = new Date(d).toLocaleDateString();
+                const hour = parseInt(d.substr(11, 2));
+                if (!dateMap.has(date)) {
+                    dateMap.set(date, []);
+                }
+                const hoursArray = dateMap.get(date);
+                if (hoursArray) hoursArray.push(hour);
+            }
+
+            setAvailabilities(dateMap);
+            setAvailableHours(dateMap.get(new Date().toLocaleDateString()));
+        }
+    }, [availabilities]);
+
+    // Checks if the day is the current day
+    const isToday = (date: number) => {
+        const today = new Date();
+        const d = new Date(year, month, date);
+
+        return today.toDateString() === d.toDateString() ? true : false;
+    };
+
+    // Sets an actual date value from the calendar number passed in
+    const selectDateValue = (day: number) => {
+        let selectedDate = new Date(year, month, day);
+
+        setDatepickerValue(() => selectedDate.toDateString());
+        setDay(day);
+        setSelectedHour(null);
+
+        if (availabilitiesMap) setAvailableHours(availabilitiesMap.get(selectedDate.toLocaleDateString()));
```

Comment:
I understand why you split the date between days and hours, so you can now get all the hours of that day.

but instead, I feel like you should just keep 1 state of available dates, and either map them to the days or hours of a day from that 1 state:

e.g.:
const [dates, setDates] = useState(<date objects>[])

// if I want to get days to show to the user
dates.map(d => d.getDay())

// if I want to show hours of that day:
dates.filter(d => d.getDay() === selectedHour).map(d => d.getHours())

---
The functions aren't probably correct, but something similar, instead of spliting the days and hours into two states.